src/compat/array/head.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { describe, expect, it } from 'vitest';
+import { head, first } from '../index';
+
+const arrayProto = Array.prototype;
+
+/**
+ * @see https://github.com/lodash/lodash/blob/6a2cc1dfcf7634fea70d1bc5bd22db453df67b42/test/head.spec.js#L1
+ */
+describe('head', () => {
+  const array = [1, 2, 3, 4];
+
+  it('should return the first element', () => {
+    expect(head(array)).toBe(1);
+  });
+
+  // it('should return `undefined` when querying empty arrays', () => {
+  //   arrayProto[0] = 1;
+  //   expect(head([])).toBe(undefined);
+  //   arrayProto.length = 0;
+  // });
```

Comment:
It seems that lodash returns undefined even if the first element of the Array prototype is overwritten so I'm not sure if we should es-toolkit's `head` should support that or have a different compat version of it?

###### Lodash's implementation
```js
function head(array) {
    return array != null && array.length ? array[0] : undefined;
}
```

Reply:
This is a catch! I think we should not handle these cases, since prototype pollution is a problem anyway 😞  Updated our docs.

https://github.com/toss/es-toolkit/commit/eee44287e7fd21e52a154adeb9a9e60729231e18